src/main/java/net/dv8tion/jda/api/JDA.java

```diff
@@ -1747,6 +1748,62 @@ default List<RichCustomEmoji> getEmojisByName(@Nonnull String name, boolean igno
         return getEmojiCache().getElementsByName(name, ignoreCase);
     }
 
+    /**
+     * Creates a new {@link ApplicationEmoji} for this bot.
+     *
+     * <p>Note that the bot is limited to 2000 Application Emojis (normal and animated).
+     *
+     * @param  name
+     *         The name for the new emoji (2-30 characters)
+     * @param  icon
+     *         The {@link Icon} for the new emoji
+     */
+    @Nonnull
+    @CheckReturnValue
+    RestAction<ApplicationEmoji> createApplicationEmoji(@Nonnull String name, @Nonnull Icon icon);
```

Comment:
This is missing documentation for the return type and exceptions

Reply:
So actually I only see this warning (https://github.com/discord/discord-api-docs/blob/main/docs/resources/Emoji.md#create-application-emoji--post-applicationsapplicationiddocs_resources_applicationapplication-objectemojis) in the docs. With a 400 Bad Request for file sizes greater then 256KiB. How exactly should I write that down?

Reply:
And I did not see any documentation that documented this in the other cases.

Reply:
I meant the exceptions in your implementation, you have several calls to `Checks` that each throw exceptions if the check fails. For instance:

```java
* @throws IllegalArgumentException
*         If null is provided or the name is not between 1-{@value CustomEmoji#EMOJI_NAME_MAX_LENGTH} characters long.
*
* @returns {@link RestAction} - Type: {@link ApplicationEmoji}
*/
```

Reply:
Yeah ok, got it. I will include those.

Reply:
No worries, I've already added it myself and some more checks and tests to round it off.